### PR TITLE
Switch to step functions

### PR DESCRIPTION
**Pull request by Belair34**
Switched all Attis functionality to use step functions created by Cybele that have the tag "Cybele". Let me know what you think

### CODE REVIEW

File: cdk/lib/attis-cdk-stack.ts [hunk 1]
```diff
@@ -21,9 +21,6 @@ export class AttisCdkStack extends Stack {
 
     const securityGroup = ec2.SecurityGroup.fromLookupByName(this, 'dev-sg', 'devvm-default-sg', vpc);
 
-    const secrets = sm.Secret.fromSecretCompleteArn(this, "attis-secrets",
```
Comment (cjudd) [inline cdk/lib/attis-cdk-stack.ts:24]: Where are the secrets moving to?

File: pom.xml [hunk 2]
```diff
@@ -10,7 +10,7 @@
 	</parent>
 	<groupId>net.javajudd.attis</groupId>
 	<artifactId>attis</artifactId>
-	<version>1.1.1-SNAPSHOT</version>
+	<version>2.0.0</version>
```
Comment (cjudd) [inline pom.xml:13]: Good catch. I am glad you did this.

File: src/main/java/net/javajudd/attis/controller/IndexController.java [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
 
 @Controller
 public class IndexController {
+
+    @Autowired
+    AWSService awsService;
+
     @GetMapping()
-    public String index() {
+    public String index(Model model) {
```
Comment (cjudd) [inline src/main/java/net/javajudd/attis/controller/IndexController.java]: Why did you add model when you are using it?

File: src/main/java/net/javajudd/attis/controller/IndexController.java [hunk 4]
```diff
@@ -1,13 +1,19 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.GetMapping;
 
 @Controller
 public class IndexController {
+
+    @Autowired
+    AWSService awsService;
```
Comment (cjudd) [inline src/main/java/net/javajudd/attis/controller/IndexController.java]: Why did you autowire this when it isn't be used? I don't think you needed to make any changes in this class at all.

File: src/main/java/net/javajudd/attis/controller/ParticipantController.java [hunk 5]
```diff
@@ -25,18 +25,20 @@ public class ParticipantController {
 
     @GetMapping({"","/"})
     public String index(Participant participant) {
-        return "participant/add-participant";
+        if(aws.getStepFunctionArn() != null) {
```
Comment (cjudd) [inline src/main/java/net/javajudd/attis/controller/ParticipantController.java]: You have this check duplicated in 2 classes. What about making this a method on AWSService? Something like aws.isStepFunctionArnNotInitialized()?

File: src/main/java/net/javajudd/attis/controller/ParticipantController.java [hunk 6]
```diff
@@ -25,18 +25,20 @@ public class ParticipantController {
...
-    public String addParticipant(@Valid Participant participant, BindingResult result, Model model) {
+    public String addParticipant(@Valid Participant participant, BindingResult result, Model model) throws InterruptedException {
         if (result.hasErrors()) {
             return "participant/add-participant";
         }
 
         participantRepository.save(participant);
-        aws.createIamUser(participant);
-        aws.createDevVM(participant);
+        aws.createUserAndVM(participant);
```
Comment (cjudd) [inline src/main/java/net/javajudd/attis/controller/ParticipantController.java]: Should this be renamed executeStepFunction?

File: src/main/java/net/javajudd/attis/service/AWSService.java [hunk 7]
```diff
@@ -1,155 +1,56 @@
...
-import software.amazon.awssdk.services.iam.model.CreateAccessKeyResponse;
-import software.amazon.awssdk.services.iam.model.CreateLoginProfileRequest;
-import software.amazon.awssdk.services.iam.model.CreateUserRequest;
-import software.amazon.awssdk.services.iam.model.CreateUserResponse;
-import software.amazon.awssdk.services.iam.model.GetUserRequest;
-import software.amazon.awssdk.services.iam.model.GetUserResponse;
-import software.amazon.awssdk.services.iam.model.Tag;
-import software.amazon.awssdk.services.iam.waiters.IamWaiter;
+import software.amazon.awssdk.services.sfn.SfnClient;
+import software.amazon.awssdk.services.sfn.model.*;
```
Comment (cjudd) [inline src/main/java/net/javajudd/attis/service/AWSService.java]: Let's not use *. If IntelliJ did this automatically, you can increase the number of classes to a larger number like 999999.

Preferences > Editor > Code Style > Java > Imports both Class count to use import and Names count to use static import.

File: src/main/java/net/javajudd/attis/service/AWSService.java [hunk 8]
```diff
@@ -1,155 +1,56 @@
...
                 .build();
 
-        IamWaiter iamWaiter = iam.waiter();
-
-        Tag name = Tag.builder().key("Participant").value(participant.getName()).build();
-        Tag company = Tag.builder().key("Company").value(participant.getCompany()).build();
-        Tag email = Tag.builder().key("Email").value(participant.getEmail()).build();
+        ListStateMachinesResponse stateMachinesResponse = client.listStateMachines();
+        List<StateMachineListItem> attisFunctions = new ArrayList<>();
+        for(StateMachineListItem function : stateMachinesResponse.stateMachines()) {
```
Comment (cjudd) [inline src/main/java/net/javajudd/attis/service/AWSService.java]: Try to use a more functional style rather than iterating with for loops.

File: src/main/java/net/javajudd/attis/service/AWSService.java [hunk 9]
```diff
@@ -1,155 +1,56 @@
...
+                }
+            }
+        }
+        return attisFunctions;
+    }
 
-        CreateUserRequest request = CreateUserRequest.builder()
-                .userName(participant.getInitials())
-                .tags(name, company, email)
+    public void createUserAndVM(Participant participant) {
```
Comment (cjudd) [inline src/main/java/net/javajudd/attis/service/AWSService.java]: Rename execute step function.

File: src/main/java/net/javajudd/attis/controller/InitializationController.java [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+    @Autowired
+    AWSService awsService;
+
+    @GetMapping
+    public String initForm(Model model) {
+        if(awsService.getStepFunctionArn() != null) {
+            return "redirect:/participant";
+        }
+        model.addAttribute("StepFunctions", awsService.getStateMachines("Cybele"));
```
Comment (cjudd) [inline src/main/java/net/javajudd/attis/controller/InitializationController.java]: StepFunctions should be lower case since it is a variable not a class.

File: cdk/lib/attis-cdk-stack.ts (hunk 1 above)
Comment (Belair34) [reply cdk/lib/attis-cdk-stack.ts:24]: The secret is now being retrieved in the email lambda that we wrote in Cybele

File: src/main/java/net/javajudd/attis/controller/IndexController.java (hunk 3 above)
Comment (Belair34) [reply src/main/java/net/javajudd/attis/controller/IndexController.java]: Good catch! I forgot to take that out when I moved everything into the initialization controller.

File: src/main/java/net/javajudd/attis/service/AWSService.java [hunk 11]
```diff
@@ -1,155 +1,55 @@
...
                 .build();
 
-        CreateUserResponse response = iam.createUser(request);
+        String participantJson = new Gson().toJson(participant);
 
-        // Wait until the user is created
-        GetUserRequest userRequest = GetUserRequest.builder()
-                .userName(response.user().userName())
+        StartExecutionRequest request = StartExecutionRequest.builder()
+                .name("AttisExecution"+UUID.randomUUID().toString().substring(0,7))
```
Comment (cjudd) [inline src/main/java/net/javajudd/attis/service/AWSService.java]: Thinking about how this will be listed in the step functions, I think it would be a good idea to include the initials in the name for traceability. Otherwise, if I have to debug, I will have to go through each and everyone to find the on associated with a user. So maybe AttisExecution-<initials>-<UUID>
